policy/scan/reporter.go

```diff
@@ -1,81 +1,19 @@
 package scan
 
 import (
-	"github.com/hashicorp/go-multierror"
 	"go.mondoo.com/cnquery/motor/asset"
 	"go.mondoo.com/cnspec/policy"
 )
 
-type Reporter interface {
-	AddReport(asset *asset.Asset, results *AssetReport)
-	AddScanError(asset *asset.Asset, err error)
-}
-
 type AssetReport struct {
 	Mrn            string
 	ResolvedPolicy *policy.ResolvedPolicy
 	Bundle         *policy.Bundle
 	Report         *policy.Report
 }
 
-type AggregateReporter struct {
-	assets           map[string]*policy.Asset
-	assetReports     map[string]*policy.Report
-	assetErrors      map[string]error
-	bundle           *policy.Bundle
-	resolvedPolicies map[string]*policy.ResolvedPolicy
-}
-
-func NewAggregateReporter(assetList []*asset.Asset) *AggregateReporter {
-	assets := make(map[string]*policy.Asset, len(assetList))
-	for i := range assetList {
-		cur := assetList[i]
-		assets[cur.Mrn] = &policy.Asset{
-			Mrn:  cur.Mrn,
-			Name: cur.Name,
-			Url:  cur.Url,
-		}
-	}
-
-	return &AggregateReporter{
-		assets:           assets,
-		assetReports:     map[string]*policy.Report{},
-		assetErrors:      map[string]error{},
-		resolvedPolicies: map[string]*policy.ResolvedPolicy{},
-	}
-}
-
-func (r *AggregateReporter) AddReport(asset *asset.Asset, results *AssetReport) {
-	r.assetReports[asset.Mrn] = results.Report
-	r.resolvedPolicies[asset.Mrn] = results.ResolvedPolicy
-
-	r.bundle = results.Bundle
-}
-
-func (r *AggregateReporter) AddScanError(asset *asset.Asset, err error) {
-	r.assetErrors[asset.Mrn] = err
-}
-
-func (r *AggregateReporter) Reports() *policy.ReportCollection {
-	errors := make(map[string]string, len(r.assetErrors))
-	for k, v := range r.assetErrors {
-		errors[k] = v.Error()
-	}
-
-	return &policy.ReportCollection{
-		Assets:           r.assets,
-		Reports:          r.assetReports,
-		Errors:           errors,
-		Bundle:           r.bundle,
-		ResolvedPolicies: r.resolvedPolicies,
-	}
-}
-
-func (r *AggregateReporter) Error() error {
-	var err error
-
-	for _, curError := range r.assetErrors {
-		err = multierror.Append(err, curError)
-	}
-	return err
+type Reporter interface {
+	AddReport(asset *asset.Asset, results *AssetReport)
+	AddScanError(asset *asset.Asset, err error)
+	Reports() *ScanResult
```

Comment:
Was this missing from the interface before? Just wondering why the interface now includes this. I saw that the reporter implementation had this.

Reply:
Yes, it was missing. Since I added different reporters now I had to properly define the interface